fontir/src/ir.rs

```diff
@@ -95,6 +98,37 @@ pub struct MiscMetadata {
     pub created: Option<DateTime<Utc>>,
 }
 
+/// IR representation of kerning.
+///
+/// In UFO terms, roughly [groups.plist](https://unifiedfontobject.org/versions/ufo3/groups.plist/)
+/// and [kerning.plist](https://unifiedfontobject.org/versions/ufo3/kerning.plist/) combined.
+#[derive(Serialize, Deserialize, Debug, Default, Clone, PartialEq, Eq)]
+#[serde(from = "KerningSerdeRepr", into = "KerningSerdeRepr")]
+pub struct Kerning {
+    pub groups: HashMap<GroupName, HashSet<GlyphName>>,
+    /// An adjustment to the space *between* two glyphs in logical order.
+    ///
+    /// Maps (side1, side2) => a mapping location:adjustment.
+    ///
+    /// Used for both LTR and RTL. The BE application differs but the concept
+    /// is the same.
+    pub kerns: HashMap<
+        (KernParticipant, KernParticipant),
+        BTreeMap<NormalizedLocation, OrderedFloat<f32>>,
+    >,
```

Comment:
this is a pretty nasty type, maybe an alias or two would be helpful?

Reply:
That IS gross, adding!